### ISSUE TITLE
Next version ... 2.0?  or 1.1?

### ISSUE THREAD

**Issue by pvretano** [opengeospatial/ogcapi-processes#292]
The purpose of this issue is to capture commentary and decisions concerning what we call the next version ... 2.0? or 1.1?  The concensus in the SWG seems to be to call it 1.1 for the following reasons:

1. The changes, although breaking, are changes to align the specification with the principles of the Spatial Data on the Web document.
2. The changes, although breaking, simplify the interaction between the client and the server and also make that interaction more "webby".
3. Calling 1.1 allows OGC to contibue with its possible plans to sync all the OGC API specification to 2.0 at some point in the near future.  It is not a given that OGC will do this but a possibility.

It should be noted that most SWG members are not hard and fast set on calling it 1.1.  The plan is to call it 1.1 for now and if, as we move forward, someone comments that is should be 2.0 we can renumber it then. 

@bpross-52n please add the motion made in the SWG to this issue.

**Comment by cportele** [comment]
I was not part of the discussion, but:

- Using 1.1 for a revision that breaks implementations is against semantic versioning, which is also part of the policies ("Minor revision releases should be 100% backwards compatible with the previous version"; "A minor revision is backward compatible with previous versions with the same major version designation").
- I hope there are no plans for force OGC API standards to a major revision!

**Comment by christophenoel** [comment]
I agree that using 1.1 would be confusing.

**Comment by bpross-52n** [comment]
Motion to set the version number of the next OGC API - Processes - Part 1: Core release to 1.1

Motion: Stan
Second: Jerome

Discussion: Peter: Technically, there are breaking changes so that, by OGC policy, makes it at 2 0.
There was no objection to unanimous consent.

**Comment by bpross-52n** [comment]
@pvretano will compare the two versions (1.0 and the new one), especially the execute request and create a table with changes. This will give us a better overview, whether we do have breaking changes.

**Comment by jerstlouis** [comment]
If there are breaking changes, I believe they only affect specific rows of the possibilities matrix, and it may be a very limited set in terms of a 1.1 server implementation also conforming to 1.0.

It might be possible for a 1.1 implementation to implement conformance to 1.0 as an extension to 1.1, e.g., allowing to recognize the `response` and `transmission` properties when conformance for both versions is declared?

I believe there is no impact for synchronous execution returning a single output.

There are additional requirements for new end-points at `/jobs/{jobId}/results/{resultId}`, but I believe nothing about that breaks conformance to 1.0 since links to those results are still included in the `/results` response and optionally in job status responses, when requested as JSON.

**Comment by jerstlouis** [comment]
If the OGC versioning scheme means that **all server implementations** conforming to 1.0 must also satisfy all 1.0 requirements, then we definitely need to call it 2.0 so as not to impose these extra 1.0 requirements and faclitate implementing 2.0.

I believe it is mostly, if not completely, possible for 1.1 **servers** to fully conform to both 1.0 and 1.1/2.0 at the same time, without a client having to specify a version number (which is not done in 1.0 requests). However, an important goal of 2.0 is to facilitate developing conforming implementations.

On the other hand, if by not breaking compatibliity, we undertand that clients sending 1.0 execution requests to 1.1 servers will always get a succesful _response_, but successful _interoperability_ will only be achieved if either:

- the server implements 1.0 backward compatibility support, or
- the client performs additional checks e.g., for `Content-Type` that will make things still work with 1.1,

then we could possibly call it 1.1.

In other words, I feel it should be mostly possible (without introducing a _"version"_ key in the payload or header) to write both:
- Clients that work with both 1.0 and 1.1/2.0 server
- Servers that work with both 1.0 and 1.1/2.0 clients

That may well not be enough to satisfy _not breaking compatibility_ per the OGC P&P definition / semantic versioning, and 2.0 is likely the sensible choice.

**Comment by pvretano** [comment]
19-AUG-2024 SWG meeting: After discussion, the SWG decided that the new version should be 2.0.  However, we are in a bit of a catch 22 here.  The only way to verify that a server implemenation can support both version 1.0 and 2.0 is to actually update the server and many implementors will not do that until 2.0 is an adopted specification.  So, we will move ahead as a 2.0 version.  Some of the SWG member (e.g. @pvretano) will continue updating their servers to support both 1.0 and 2.0 and if any issues arrise, new issues will be created to address to conflicts.  Closing this for now.

**Comment by pvretano** [comment]
19-AUG-2024: The editors will create a PR updating the version of the document to 2.0.

**Comment by pvretano** [comment]
Version updated to 2.0.  Closing.
